packages/azure-http-specs/specs/azure/client-generator-core/hierarchy-building/main.tsp

```diff
@@ -0,0 +1,111 @@
+import "@typespec/http";
+import "@typespec/spector";
+import "@azure-tools/typespec-client-generator-core";
+
+using Http;
+using Spector;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for @hierarchyBuilding decorator.")
+@scenarioService("/azure/client-generator-core/hierarchy-building")
+namespace _Specs_.Azure.ClientGenerator.Core.HierarchyBuilding;
+
+@discriminator("kind")
+model Animal {
+    @doc("The kind of animal")
+    kind: string;
+    
+    @doc("Name of the animal")
+    name: string;
+  }
+
+  alias PetContent = {
+    @doc("Whether the pet is trained")
+    trained: boolean;
+  };
+
+  model Pet extends Animal {
+    kind: "pet";
+    ...PetContent;
+  }
+
+  alias DogContent = {
+    @doc("The breed of the dog")
+    breed: string;
+  };
+
+  @global.Azure.ClientGenerator.Core.Legacy.hierarchyBuilding(Pet)
+  model Dog extends Animal {
+    kind: "dog";
+    ...PetContent;
+    ...DogContent;
+  }
+
+interface AnimalOperations {
+  @scenario
+  @scenarioDoc("""
+    Test operation that accepts Animal input and returns Animal output.
+    This validates basic polymorphic behavior in the inheritance hierarchy.
+    """)
```

Comment:
We shall add "expected request body" and "expected response body" in scenarioDoc string according to definition in mockapi.ts so that developers could know how to test easily. Here is example: https://github.com/Azure/typespec-azure/blob/6df66fbb0d94794122d75bcfd50be6d9403439f8/packages/azure-http-specs/specs/azure/resource-manager/common-properties/managed-identity.tsp#L69-L103